design/client-policies.md

```diff
@@ -207,16 +224,472 @@ Conditions and executors of a policy should be evaluated on endpoints where a cl
+Policies, profiles, conditions, executors can be configured on Admin Console. To do so, add "Client Policies" tab on Realm->Realm Settings, which means an administrator can have client policies per realm.
+
+An administrator can create new profile and add executors.
+
+An administrator can create new policy, add conditions and refer to profiles.
 
-An administrator can create new policy and add conditions and executors.
-An administrator can update their created policies by add/modify/delete conditions and executors while the administrator can not update pre-set policies.
+An administrator can update their created policies and profiles while they can not update built-in policies and profiles.
 
 ### Backward Compatibility
 
-Client Policies can take over Client Registration. However, Client Registration Policies also still co-exist. In the future, Client Registration Policies will be removed and the existing client registration policies will be migrated into new client policies automatically.
+Client Policies can take over Client Registration Policies. However, Client Registration Policies also still co-exist. In the future, Client Registration Policies will be removed and the existing client registration policies will be migrated into new client policies automatically.
+
+The RH-SSO has already supported Client Registration Policies officially, not tech preview while it will treat newly introduced Client Policies as tech preview at first. Therefore, the keycloak will include both Client Registration Policies and Client Policies. After Client Policies being supported officially by RH-SSO, the keycloak drops Client Registration Policies.
 
 ### Extensibility
 
-Condition, Executor and Policy are implemented as Provider. Therefore, developers can implement their own condition, executor and policy by themselves, which makes it easy to realize some security profiles that may come in the future.
+Condition, Executor, Profile and Policy are implemented as Provider. Therefore, developers can implement their own condition, executor, profile and policy by themselves, which makes it easy to realize some security profiles that may come in the future.
+
+## External Interface - Admin REST API
+
+To add/read/modify/delete conditions/executors/policies/profiles, following Admin REST APIs are supported.
+
+The style of API description conforms to [keycloak's API Documentation](https://www.keycloak.org/docs-api/11.0/rest-api/index.html).
+
+### Design Concept
+
+Admin REST API for Client Policies are designed to satisfy the following requirements.
+
+- keycloak admin can view and update the client policy on the editable textarea UI (editing JSON representation directly).
+- keycloak admin can control policies, conditions and executors on the ordinal UI (e.g. like the existing Client Registration Policy).
```

Comment:
Just a note, that there is a question how to implement these points because the new admin console is in the design right now, but it is not yet ready... So if you want to start implementation of these points, you may not yet be able to add support in the admin console UI of new admin console as it is not ready yet... 

If you implement some support to the old admin console, I suggest to add here only the editing by JSON to be able to test client policies with the old admin console.

Reply:
@mposolda Thank you for your comment.

AFAIK, the situation for the New Admin Console are as follows.

- The New Admin Console has not yet appeared on the latest keycloak.
- The UI design for Client Policies on the New Admin Console has not yet been fixed. It has been pending because the design for the New Admin REST API for Client Policies had not yet been approved and merged.

Considering these points, I would like to work on the support of New Admin REST API as follows.

1. Resume discussion on the UI design for Client Policies on the New Admin Console (https://marvelapp.com/prototype/6e70eh2/screen/74918976) and fix it.
2. In parallel, Implement the New Admin REST API for Client Policies by following this merged design document.
3. After completing 1 and 2, if the New Admin Console appears on the keycloak's master branch, implement the UI for Client Policies on the New Admin Console. If not, implement the UI for Client Policies on the existing Admin Console (only editable JSON).

Reply:
@tnorimat Your proposal sounds great to me. Thanks!

Reply:
@tnorimat FYI.  The new admin console is being developed in its own repo: https://github.com/keycloak/keycloak-admin-ui

I agree with @mposolda that your proposal is good.  If you want this functionality soon then it will need to go into old admin console.

Reply:
@ssilvert Thank you for your comment. I would like to check https://github.com/keycloak/keycloak-admin-ui .